test/IntegrationTests/MSTest.VstestConsoleWrapper.IntegrationTests/MSTest.VstestConsoleWrapper.IntegrationTests.csproj

```diff
@@ -2,14 +2,26 @@
 
   <PropertyGroup>
     <TargetFramework>$(NetFrameworkMinimum)</TargetFramework>
-    <UseInternalTestFramework>true</UseInternalTestFramework>
-    <!-- Don't generate MSTest using -->
-    <UseMSTest>false</UseMSTest>
+    <EnableMSTestRunner>true</EnableMSTestRunner>
+    <UseMSTestFromSource>true</UseMSTestFromSource>
     <OutputType>Exe</OutputType>
   </PropertyGroup>
 
   <ItemGroup>
     <ProjectReference Include="$(RepoRoot)test\Utilities\Automation.CLI\Automation.CLI.csproj" />
+    <ProjectReference Include="$(RepoRoot)src\Adapter\MSTest.TestAdapter\MSTest.TestAdapter.csproj" />
```

Comment:
I guess the original reason why we had an internal test framework is that we don't want to use whatever we have from source for testing. Because if it's broken in some way, then the tests may be testing what it should.

Reply:
Although what you say is a valid concern, I think we have better ways around that (min number of tests, in the past we had a txt file that was listing all tests passing and was checked at the end of the session).

The original reason was around around modification of the states of MSTest internal and that's why I think we don't have much way around for unit tests.

As we discussed in past, the only viable option would be to rely on NuGet packages of the last insertion (or last daily) but we would still need to find a way to isolation classes from MSTest local to MSTest package.